api/all/src/main/java/io/opentelemetry/api/internal/OtelEncodingUtils.java

```diff
@@ -37,6 +38,14 @@
     return decoding;
   }
 
+  private static boolean[] buildValidHexArray() {
+    boolean[] validHex = new boolean[Character.MAX_VALUE];
```

Comment:
There are times when I hate java. `Character.MAX_VALUE` is a `char`, which somehow you can use as an `int` because we can't have nice things. :(